x/datadeal/genesis_test.go

```diff
@@ -0,0 +1,83 @@
+package datadeal_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/medibloc/panacea-core/v2/types/assets"
+	"github.com/medibloc/panacea-core/v2/types/testsuite"
+	"github.com/medibloc/panacea-core/v2/x/datadeal"
+	"github.com/medibloc/panacea-core/v2/x/datadeal/types"
+	"github.com/stretchr/testify/suite"
+	"github.com/tendermint/tendermint/crypto/secp256k1"
+)
+
+type genesisTestSuite struct {
+	testsuite.TestSuite
+}
+
+var (
+	acc1         = sdk.AccAddress(secp256k1.GenPrivKey().PubKey().Address())
+	defaultFunds = sdk.NewCoins(sdk.NewCoin(assets.MicroMedDenom, sdk.NewInt(10000000000)))
+)
+
+func TestGenesisTestSuite(t *testing.T) {
+	suite.Run(t, new(genesisTestSuite))
+}
+
+func makeSampleDeal(dealID uint64) types.Deal {
+	return types.Deal{
+		Id:           dealID,
+		Address:      types.NewDealAddress(dealID).String(),
+		DataSchema:   nil,
+		Budget:       &sdk.Coin{Denom: "umed", Amount: sdk.NewInt(1000000)},
+		MaxNumData:   1000,
+		CurNumData:   0,
+		BuyerAddress: acc1.String(),
+		Status:       0,
+	}
+}
+
+func (suite *genesisTestSuite) TestInitGenesis() {
+	deal1 := makeSampleDeal(1)
+	deal2 := makeSampleDeal(2)
+
+	genesis := types.GenesisState{
+		Deals:          []types.Deal{deal1, deal2},
+		NextDealNumber: 3,
+	}
+
+	datadeal.InitGenesis(suite.Ctx, suite.DataDealKeeper, genesis)
+
+	getDeal1, err := suite.DataDealKeeper.GetDeal(suite.Ctx, 1)
+	suite.Require().NoError(err)
+	suite.Require().Equal(genesis.Deals[0], getDeal1)
+
+	getDeal2, err := suite.DataDealKeeper.GetDeal(suite.Ctx, 2)
+	suite.Require().NoError(err)
+	suite.Require().Equal(genesis.Deals[1], getDeal2)
+
+}
+
+func (suite *genesisTestSuite) TestExportGenesis() {
+	deal1 := makeSampleDeal(1)
+	deal2 := makeSampleDeal(2)
+
+	genesis := types.GenesisState{
+		Deals:          []types.Deal{deal1},
+		NextDealNumber: 2,
+	}
+
+	datadeal.InitGenesis(suite.Ctx, suite.DataDealKeeper, genesis)
+
+	err := suite.FundAccount(suite.Ctx, acc1, defaultFunds)
+	suite.Require().NoError(err)
+
+	_, err = suite.DataDealKeeper.CreateDeal(suite.Ctx, acc1, deal2)
+	suite.Require().NoError(err)
+
+	genesisStatus := datadeal.ExportGenesis(suite.Ctx, suite.DataDealKeeper)
+	suite.Require().Equal(deal1.Address, genesisStatus.Deals[0].Address)
+	suite.Require().Equal(deal2.Address, genesisStatus.Deals[1].Address)
+	suite.Require().Equal(uint64(3), genesisStatus.NextDealNumber)
+}
```

Comment:
I think you should write comparing the all of the fields of deal.

Reply:
Okay, I fix it. Thanks!